meson.build

```diff
@@ -63,6 +63,11 @@ config_host = configuration_data()
 config_host.set('NVME_AQ_QSIZE', get_option('aq_qsize'),
   description: 'admin command queue size')
 
+config_host.set('HAVE_LINUX_IOMMUFD_H',
+  cc.has_header_symbol('linux/iommufd.h', 'IOMMU_IOAS_ALLOC')
+  and cc.has_header_symbol('linux/vfio.h', 'VFIO_DEVICE_BIND_IOMMUFD'),
+  description: 'weather IOMMU_IOAS_ALLOC is defined in linux/iommufd.h')
+
```

Comment:
Actually you are not testing if "have linux iommufd.h" - you are testing if `iommufd.h` has `IOMMU_IOAS_ALLOC`. I'm not sure if it makes sense to have a `HAVE_LINUX_IOMMUFD_H` that just checks with `cc.check_header()` - or we should just rename this to `HAVE_IOMMU_IOAS_ALLOC`.

Reply:
I actually went with HAVE_VFIO_DEVICE_BIND_IOMMUFD. I explain this in the commit message. It basically boils down to the fact that IOMMU_IOAS_ALLOC and the linux/iommufd.h file was present before the patch that activated the iommufd system (it was probably included as preparation). So the think that marks what is needed is in fact the BIND_IOMMUFD ioctl from vfio.